core/frontend/src/internal/tile/IModelTileTree.ts

```diff
@@ -513,21 +495,52 @@ export class IModelTileTree extends TileTree {
     }
 
     const relevantChange = changes.find((c) => c.timeline.modelId === this.modelId);
+    if (!relevantChange || relevantChange.elements.size === 0)
+      return;
+
+    const elementIds = Array.from(relevantChange.elements);
+    if (elementIds.length === 0)
+      return;
+
+    let placements: (Placement2d | Placement3d | undefined)[] = [];
+    try {
+      placements = await this.iModel.elements.getPlacements(elementIds);
+    } catch {
+      placements = new Array(elementIds.length).fill(undefined);
+    }
+
     const changedElements: ElementGeometryChange[] = [];
 
-    if (relevantChange) {
-      for (const id of relevantChange.elements) {
-        changedElements.push({
-          id,
-          type: DbOpcode.Update,
-          range: await this.getElementRange(id),
-        });
+    for (let i = 0; i < elementIds.length; i++) {
+      const id = elementIds[i];
+      const placement = placements[i];
+      let range = Range3d.createNull();
+
+      try {
+        if (placement) {
+          if (isPlacement3dProps(placement)) {
+            range = Placement3d.fromJSON(placement).calculateRange();
+          } else if (isPlacement2dProps(placement)) {
+            range = Placement2d.fromJSON(placement).calculateRange();
+          }
+        }
+      } catch {
+        range = Range3d.createNull();
       }
+
+      changedElements.push({
+        id,
+        type: DbOpcode.Update,
+        range,
+      });
```

Comment:
`placements` is already an array of `Placement`s (plus element Id), you don't need to do `fromJSON`. Something like this:

```suggestion
    changedElements = placements.map((x) => {
      return {
        id: x.elementId,
        type: DbOpcode.Update,
        rance: x.calculateRange(),
      };
    });
```

Reply:
<img width="800" height="91" alt="image" src="https://github.com/user-attachments/assets/e9fc2dd3-f13c-4362-b63f-345d228d589e" />

It will have this error

Reply:
I pushed this change, no such error. (There was a typo `rance` for `range` though).

Reply:
That makes perfect sense, I was destructuring with `map(({ elementId, ...placement }) => ...)`, which ended up stripping away the class type, so placement became a plain object and lost access to methods like `calculateRange`. This was super helpful!